eng/build.ps1

```diff
@@ -109,3 +110,36 @@ if ($env:TreatWarningsAsErrors -eq 'false') {
 
 Write-Host "& `"$PSScriptRoot/common/build.ps1`" $arguments"
 Invoke-Expression "& `"$PSScriptRoot/common/build.ps1`" $arguments"
+$buildExitCode = $LASTEXITCODE
+
+# Install MAUI workload after restore if -restore-maui was passed
+# Only on Windows and macOS (MAUI doesn't support Linux)
+$restoreMauiPassed = $properties -contains "-restore-maui"
+$isWindowsOrMac = ($IsWindows -or $IsMacOS -or (-not (Get-Variable -Name IsWindows -ErrorAction SilentlyContinue)))
+if ($restoreMauiPassed -and $buildExitCode -eq 0 -and $isWindowsOrMac) {
+  Write-Host ""
+  Write-Host "Installing MAUI workload into local .dotnet..."
+  
+  $repoRoot = Split-Path $PSScriptRoot -Parent
+  $dotnetRoot = Join-Path $repoRoot ".dotnet"
```

Comment:
This will not always be true, as there are cases where the machine-wide version matches the version in global.json then the build won't create a local .dotnet folder. Instead, this script should be calling the `dotnet.cmd` or `dotnet.sh` scripts in the root of the repo which will resolve to the right place always.

cc: @jfversluis